pkg/cli/admin/release/extract_tools.go

```diff
@@ -908,3 +954,121 @@ func copyAndReplace(errorOutput io.Writer, w io.Writer, r io.Reader, bufferSize
+	}
+
+	installConfigBytes, err := os.ReadFile(installConfigPath)
+	if err != nil {
+		return config, err
+	}
+
+	data := installConfig{}
+
+	if err := yaml.Unmarshal(installConfigBytes, &data); err != nil {
+		return config, fmt.Errorf("failed to parse %s: %w", installConfigPath, err)
+	}
+
+	if data.APIVersion != "v1" {
+		return config, fmt.Errorf("unrecognized %s API version: %q (expected %q)", installConfigPath, data.APIVersion, "v1")
+	}
+
+	config.RequiredFeatureSet = pointer.String(string(data.FeatureSet))
+	config.Profile = pointer.String(manifest.DefaultClusterProfile) // assumption, but there's no install-config data about profile to give us more insight
+	for key := range data.Platform {
+		config.Platform = pointer.String(key)
+	}
+
+	if data.Capabilities != nil {
+		config.Capabilities = &configv1.ClusterVersionCapabilitiesStatus{}
+		if enabled, ok := configv1.ClusterVersionCapabilitySets[data.Capabilities.BaselineCapabilitySet]; !ok {
+			return config, fmt.Errorf("unrecognized baselineCapabilitySet %q", data.Capabilities.BaselineCapabilitySet)
+		} else {
+			if data.Capabilities.BaselineCapabilitySet == configv1.ClusterVersionCapabilitySetCurrent {
+				klog.Infof("If the eventual cluster will not be the same minor version as this %s 'oc', the actual %s capability set may differ.", reportedVersion, data.Capabilities.BaselineCapabilitySet)
+			}
+			config.Capabilities.EnabledCapabilities = append(config.Capabilities.EnabledCapabilities, enabled...)
+		}
+		config.Capabilities.EnabledCapabilities = append(config.Capabilities.EnabledCapabilities, data.Capabilities.AdditionalEnabledCapabilities...)
+
+		klog.Infof("If the eventual cluster will not be the same minor version as this %s 'oc', the known capability sets may differ.", reportedVersion)
+		config.Capabilities.KnownCapabilities = configv1.KnownClusterVersionCapabilities
+	}
+
+	return config, nil
+}
+
+func findClusterIncludeConfig(ctx context.Context, restConfig *rest.Config) (manifestInclusionConfiguration, error) {
+	config := manifestInclusionConfiguration{}
+
+	client, err := configv1client.NewForConfig(restConfig)
+	if err != nil {
+		return config, err
+	}
+
+	if featureGate, err := client.FeatureGates().Get(ctx, "cluster", metav1.GetOptions{}); err != nil {
+		return config, err
+	} else {
+		config.RequiredFeatureSet = pointer.String(string(featureGate.Spec.FeatureSet))
+	}
+
+	if clusterVersion, err := client.ClusterVersions().Get(ctx, "version", metav1.GetOptions{}); err != nil {
+		return config, err
+	} else {
+		config.Overrides = clusterVersion.Spec.Overrides
+		config.Capabilities = &clusterVersion.Status.Capabilities
+
+		// FIXME: eventually pull in GetImplicitlyEnabledCapabilities from https://github.com/openshift/cluster-version-operator/blob/86e24d66119a73f50282b66a8d6f2e3518aa0e15/pkg/payload/payload.go#L237-L240 for cases where a minor update would implicitly enable some additional capabilities.  For now, 4.13 to 4.14 will always enable MachineAPI.
```

Comment:
Are we not doing it because we need to move this code to API code first then change it here?